diff --git a/one.js b/two.js
index 5f964fe..5815d15 100644
--- a/one.js
+++ b/two.js
@@ -25,10 +25,8 @@ class Cargo {
   }
 
   move({ count, from, to }) {
-    for (let i = 0; i < count; i++) {
-      const item = this.stacks[from - 1].shift();
-      this.stacks[to - 1].unshift(item);
-    }
+    const items = this.stacks[from - 1].splice(0, count);
+    this.stacks[to - 1].splice(0, 0, ...items);
   }
 
   getTopLayer() {
