diff --git a/one.js b/two.js
index 924b903..11a9f7b 100644
--- a/one.js
+++ b/two.js
@@ -16,17 +16,25 @@ for (let char = ACode; char <= ZCode; char++) {
 }
 
 let sum = 0;
-for await (const [first, second] of readInput(import.meta.url, splitInput)) {
-  const firstSet = new Set(first);
-  for (const char of second) {
-    if (firstSet.has(char)) {
+for await (const [one, two, three] of readByThree()) {
+  const first = new Set(one);
+  const second = new Set(two);
+  const third = new Set(three);
+  for (const char of first) {
+    if (second.has(char) && third.has(char)) {
       sum += priorities[char];
-      break;
     }
   }
 }
 console.log(sum);
 
-function splitInput(input) {
-  return [input.slice(0, input.length / 2), input.slice(input.length / 2)];
+async function* readByThree() {
+  let group = [];
+  for await (const line of readInput(import.meta.url)) {
+    group.push(line);
+    if (group.length === 3) {
+      yield group;
+      group = [];
+    }
+  }
 }
