diff --git a/one.js b/two.js
index 75a1c5d..37efc5d 100644
--- a/one.js
+++ b/two.js
@@ -4,14 +4,8 @@ import { readInput } from '../utils.js';
 
 let count = 0;
 for await (const { first, second } of readInput(import.meta.url, splitInput)) {
-  const firstLength = getRangeLength(first);
-  const secondLength = getRangeLength(second);
   const overlap = getOverlap(first, second);
-  if (
-    overlap &&
-    (getRangeLength(overlap) === firstLength ||
-      getRangeLength(overlap) === secondLength)
-  ) {
+  if (overlap) {
     count++;
   }
 }
@@ -30,10 +24,6 @@ function getOverlap(first, second) {
   };
 }
 
-function getRangeLength(range) {
-  return range.end - range.start + 1;
-}
-
 function splitInput(input) {
   const [first, second] = input.split(',');
   return {
